tracee/tracee.go

```diff
@@ -831,9 +831,15 @@ func (t *Tracee) initBPF(bpfObjectPath string) error {
 			case kprobe:
 				// todo: after updating minimal kernel version to 4.18, use without legacy
 				_, err = prog.AttachKprobeLegacy(probe.event)
+				if err != nil {
+					_, err = prog.AttachKprobe(probe.event)
```

Comment:
I think if it's better to do the reverse: try the new `AttachKprobe` and if it fails, fallback to `AttachKprobeLegacy`. In this way, the program would correctly be detached when the process terminates and it will help compatibility with other programs using the debugfs based kprobe, such as gobpf-based programs. See also: https://github.com/iovisor/gobpf/issues/223

And it would be more aligned on [the way bcc is doing it](https://github.com/iovisor/bcc/blob/6e9b4509fc7a063302b574520bac6d49b01ca97e/src/cc/libbpf.c#L1021-L1027).

You could also add a comment to explain why `AttachKprobeLegacy` might fail, such as
```
// Legacy kprobes have a system-wide name (chosen by libbpfgo.go poke_kprobe_events())
// that might conflict with other BPF programs unrelated to tracee. New perf based kprobes
// don't have this problem but require Linux 4.17.
```

Reply:
We have discussed this with @yanivagman in the previous PR and per his recommendation I have done it this way. But I can do it in reverse of course.

Reply:
@ogozman that's correct. In the original PR, you replaced AttachKprobeLegacy with AttachKprobe, so I mentioned that we need to support the legacy kprobe attach mechanism as we still support kernels 4.14-4.18.
I know that I wrote that we should default to the legacy attach mechanism, but it actually makes more sense to do the reverse, like @alban suggested.
Anyways, we are planning to bump the minimal kernel version to 4.18 soon, so that's only temporary.